### PR TITLE
spike: Sync errors

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Messing around with fixes for some of the syncing errors we're coming across.

**Compilation errors**
- [x] Adds `base_goerli` and `base` to the chain spec and genesis value parser in the CLI.
- [x] Enables `reth-payload-builder`'s `optimism` feature flag in the basic payload builder.

**Implementation errors**
- [ ] Fixes `no_tx_pool`
    - Currently, it is implemented in such a way that it blocks the job entirely, no payload is built. The idea seems to be to skip the polling job entirely and build the payload upfront. A hacky fix is here, will clean up once we fix the sync errors.
- [ ] Fixes `compute_pending_block`
    -  `compute_pending_block` is implemented incorrectly atm - it's blocking the payload job from resolving. Removed because it's not consensus critical, but needs investigation.
- [x] Fixes `TxDeposit` RLP encoding / decoding.
    - Was out of spec. See [here](https://github.com/ethereum-optimism/optimism/blob/develop/specs/deposits.md#the-deposited-transaction-type).
- [x] Fixes extra data in optimism's payload builder.
    - Dumb error; Wasn't returning empty bytes when optimism is enabled in chainspec. 
- [x] Fix revm env 
    - Pre-regolith, the block gas limit and base fee should be entirely ignored for deposit txs. This can be done via revm's `optional_block_gas_limit` and `optional_no_base_fee` feature flags.
- [x] Fix `PayloadAttributes` `gas_limit` deserialization.
    - It is sent as a hex string (ex. `0x0a`), so we needed to add a custom deserializer.
- [x] Fixed `L1BlockInfo` tx calldata offsets for the L1 fee calculation.

### CODE REVIEW

File: crates/primitives/src/chain/spec.rs [hunk 1]
```diff
@@ -292,6 +292,7 @@ pub static BASE_GOERLI: Lazy<Arc<ChainSpec>> = Lazy::new(|| {
                 Hardfork::Paris,
                 ForkCondition::TTD { fork_block: Some(0), total_difficulty: U256::from(0) },
             ),
+            (Hardfork::Bedrock, ForkCondition::Block(0)),
```
Comment (roberto-bayardo) [inline crates/primitives/src/chain/spec.rs:295]: Can you add this to the BASE_MAINNET config below too?

File: crates/payload/basic/src/lib.rs [hunk 2]
```diff
@@ -173,6 +173,39 @@ where
         let until = tokio::time::Instant::now() + self.config.deadline;
         let deadline = Box::pin(tokio::time::sleep_until(until));
 
+        #[cfg(feature = "optimism")]
+        if config.attributes.no_tx_pool {
```
Comment (clabby) [inline crates/payload/basic/src/lib.rs:177]: This is not what we'll need in the end, but should be around the right place. Will clean up impl once we get past the payload / execution errors.

File: crates/payload/basic/src/lib.rs [hunk 3]
```diff
@@ -371,15 +404,7 @@ where
...
-                #[cfg(not(feature = "optimism"))]
-                {
-                    this.pending_block = Some(PendingPayload { _cancel, payload: rx });
-                }
-
-                #[cfg(feature = "optimism")]
-                if this.config.compute_pending_block {
-                    this.pending_block = Some(PendingPayload { _cancel, payload: rx });
-                }
+                this.pending_block = Some(PendingPayload { _cancel, payload: rx });
```
Comment (clabby) [inline crates/payload/basic/src/lib.rs:407]: TODO: Find the right place for `compute_pending_block`.

File: crates/revm/src/executor.rs [hunk 4]
```diff
@@ -240,16 +240,29 @@ where
...
+                        block_available_gas,
+                    }
+                    .into())
+                }
+
+                if transaction.is_deposit() && !is_regolith {
+                    self.evm.env.cfg.disable_base_fee = true;
+                    self.evm.env.cfg.disable_block_gas_limit = true;
+                }
+
```
Comment (clabby) [inline crates/revm/src/executor.rs:265]: L364 - account for pre-regolith system txs consuming 0 gas.

**Review by refcell — APPROVED** [review]

**Pull request merged**